shell/app/common/components/pagination/index.tsx

```diff
@@ -0,0 +1,148 @@
+//
+// This program is free software: you can use, redistribute, and/or modify
+// it under the terms of the GNU Affero General Public License, version 3
+// or later ("AGPL"), as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import React from 'react';
+import { Popover, Input, Button, Dropdown, Menu } from 'antd';
+import { ErdaIcon } from 'common';
+import i18n from 'i18n';
+import { PAGINATION } from 'app/constants';
+
+import './index.scss';
+
+interface IPaginationProps {
+  total?: number;
+  current?: number;
+  pageSize?: number;
+  pageSizeOptions?: string[];
+  showSizeChanger?: boolean;
+  onChange?: (page: number, pageSize?: number) => void;
+}
+
+export interface IPaginationJumpProps {
+  pagination: IPaginationProps;
+  hiddenPopover: () => void;
+}
+
+const Pagination = (pagination: IPaginationProps) => {
+  const {
+    total = 0,
+    current = 1,
+    pageSize = PAGINATION.pageSize,
+    pageSizeOptions = PAGINATION.pageSizeOptions,
+    showSizeChanger = true,
+    onChange,
+  } = pagination;
+
+  const [goToVisible, setGoToVisible] = React.useState(false);
+
+  const paginationCenterRender = (
+    <Popover
+      content={<PaginationJump pagination={pagination} hiddenPopover={() => setGoToVisible(false)} />}
+      trigger="click"
+      getPopupContainer={(triggerNode) => triggerNode.parentElement as HTMLElement}
+      placement="top"
+      overlayClassName="pagination-jump"
+      visible={goToVisible}
+      onVisibleChange={setGoToVisible}
+    >
+      <div className="pagination-center bg-hover px-3 rounded cursor-pointer" onClick={() => setGoToVisible(true)}>
+        {total ? pagination.current : 0} / {(total && pageSize && Math.ceil(total / pageSize)) || 0}
+      </div>
+    </Popover>
+  );
+
+  const pageSizeMenu = (
+    <Menu>
+      {(pageSizeOptions || [])
+        .filter((item) => item)
+        .map((item: string | number) => {
+          return (
+            <Menu.Item key={item} onClick={() => onChange?.(1, +item)}>
+              <span className="fake-link mr-1">{i18n.t('{size} items / page', { size: item })}</span>
+            </Menu.Item>
+          );
+        })}
+    </Menu>
+  );
+
+  return (
+    <div className="erda-pagination flex justify-end items-center">
+      <div className="erda-pagination-total mr-2">共计 {total} 条</div>
+      <div className="erda-pagination-content inline-flex">
+        <div
+          className={`bg-hover p-2 leading-none ${current === 1 ? 'disabled' : ''}`}
+          onClick={() => current > 1 && onChange?.(current - 1)}
+        >
+          <ErdaIcon type="left" size={18} color="currentColor" />
```

Comment:
currentColor is unpredictable if there is no parent color specified. I dont recommend to use it